custom_connectors/oauth2/zoho_mail_connector.rb

```diff
@@ -0,0 +1,540 @@
+          { name: 'Bcc',
+            hint: 'Comma separated value of valid email addresses' },
+          { name: 'Body Type',
+            control_type: 'select',
+            pick_list: [
+              %w[html html],
+              %w[plaintext plaintext]
+            ] }
+        ]
+      end,
+      execute: lambda { |connection, input|
+        account_id = if input['Account'].blank?
+                       call(:get_account_id, connection)
+                     else
+                       input['Account']
+                     end
+        payload = {
+          'fromAddress' => input['From'],
+          'toAddress' => input['To'],
+          'subject' => input['Subject'],
+          'content' => input['Content'],
+          'ccAddress' => input['Cc'],
+          'bccAddress' => input['Bcc'],
+          'mailFormat' => input['Body Type'],
+          'mode' => 'draft'
+        }
+        post("/api/accounts/#{account_id}/messages", payload)['data']
+      },
+      sample_output: lambda do |_connection, input|
+        {
+          fromAddress: input['From'],
+          subject: input['Subject'],
+          toAddress: input['To'],
+          content: input['Content'],
+          ccAddress: input['Cc'],
+          bccAddress: input['Bcc']
+        }
+      end,
+
+      output_fields: lambda { |object_definitions|
+                       object_definitions['send_mail_response']
+                     }
+    },
+
+    create_task: {
+      description: "Create <span class='provider'>New task</span> in " \
+        "<span class='provider'>Zoho Mail</span>",
+      input_fields: lambda do
+        [
+          { name: 'Account', control_type: 'select',
+            pick_list: 'accounts', optional: true,
+            hint: 'Select the account (Zoho account/ POP account)
+            from the list of accounts available in Zoho Mail.' },
+          { name: 'From', control_type: 'select',
+            pick_list: 'from_address', optional: false },
+          { name: 'To', optional: false, control_type: 'email',
+            hint: 'Assignee' },
+          { name: 'Subject', optional: false },
+          { name: 'Content' }
+
+        ]
+      end,
+      execute: lambda { |connection, input|
+                 account_id = if input['Account'].blank?
+                                call(:get_account_id, connection)
+                              else
+                                input['Account']
+                              end
+                 payload = {
+                   'fromAddress' => input['From'],
+                   'toAddress' => input['To'].gsub('@', '+task@'),
+                   'subject' => input['Subject'],
+                   'content' => input['Content'],
+                   'ccAddress' => input['Cc'],
+                   'bccAddress' => input['Bcc'],
+                   'mailFormat' => input['Body Type']
+                 }
+                 post("api/accounts/#{account_id}/messages", payload)['data']
```

Comment:
This endpoint seems to differ from possible endpoints to create tasks. 
https://www.zoho.com/mail/help/api/post-add-new-task.html 
or
https://www.zoho.com/mail/help/api/post-group-task.html

is there a reason for this?

Reply:
There is one more way to create tasks. Since I have already used a send mail API, I chose to re-use them.
https://help.zoho.com/portal/kb/articles/how-do-i-assign-a-task-to-myself-by-sending-an-email

Reply:
Hi, when we tested the `create task` action, it seems that there is a significant delay before the task is created in ZohoMail. Our testers report seeing the task in the UI after approximately 2 days. Is this normal?

Reply:
Hi @bennettgo , it is not a normal behavior.
The task will be created instantly. I'm unable to reproduce the issue from my account. I tried it and it worked.
Can you ask them to see the section "Created By Me" in tasks and hit refresh once the action is complete. Please try this and If issue persists, do let me know.

Reply:
Sharing a recipe for reference.
https://www.workato.com/recipes/971837-new-note-in-evernote-will-create-new-task-in-zoho-mail#recipe

Reply:
Hi @bennettgo are there any updates on this?

Reply:
Hello Team, are they waiting for our ETA?

Reply:
Hi Team! Just checking if you have signed the developer agreement with Workato? @zohomail

Reply:
Yes drrcked61, signed with the account linked to zmintegration@zohomail.com. Please verify

Reply:
Thanks Team! @bennettgo  may we know how do we confirm this?

Reply:
Team, any updates on this?